### PR TITLE
ARROW-12780: [CI][C++] Install necessary packages for MinGW builds

### PR DESCRIPTION
**Pull request by lidavidm**
libxml2 is now required to run clang on MinGW. This can be found by `strace`ing clang under MinGW; a dialog will pop up telling you the name of any missing libraries (ldd, notably, may not give you this info).

### CODE REVIEW

File: ci/scripts/msys2_setup.sh [hunk 1]
```diff
@@ -34,6 +34,7 @@ case "${target}" in
     packages+=(${MINGW_PACKAGE_PREFIX}-gflags)
     packages+=(${MINGW_PACKAGE_PREFIX}-grpc)
     packages+=(${MINGW_PACKAGE_PREFIX}-gtest)
+    packages+=(${MINGW_PACKAGE_PREFIX}-libxml2)
     packages+=(${MINGW_PACKAGE_PREFIX}-libutf8proc)
```
Comment (kou) [inline ci/scripts/msys2_setup.sh:38]: Could you fix order?

```suggestion
    packages+=(${MINGW_PACKAGE_PREFIX}-libutf8proc)
    packages+=(${MINGW_PACKAGE_PREFIX}-libxml2)
```

**Review by kou — APPROVED** [review]
+1

Wow! I didn't know the `strace` trick!

**Comment by kou** [comment]
Could you also improve the title?

**Comment by lidavidm** [comment]
Title & order fixed, thank you!

I kicked Travis again, though it seems unrelated, and the integration build should be fixed by #10314.

**Comment by kou** [comment]
Thanks!

**Pull request closed without merging**